File: csharp/CSharpExample/Examples/SimpleOrderExample.cs
using ATG.API.Types;
using ATG.API.Types.Requests;
using ATG.API.Types.Strategies;
using System.Text.Json;
using System.Net.Http.Json;
using CSharpExample.Utils;

namespace CSharpExample.Examples
{
    class SimpleOrderExample
    {
        private readonly HttpClient _httpClient;
        private readonly Config _config;

        public SimpleOrderExample(HttpClient client, Config config)
        {
            _httpClient = client;
            _config = config;
        }

        public async Task RunAsync()
        {
            try
            {
                Console.WriteLine("*** RUNNING SIMPLE ORDER EXAMPLE ***");
                Console.WriteLine();

                string strategyId = await NewSimpleOrder();
                await Task.Delay(1000);

                var simpleOrder = await GetSimpleOrderById(strategyId);
                var status = simpleOrder.Status;
                await Task.Delay(500);

                await UpdateSimpleOrder(strategyId, status);
                await Task.Delay(500);

                await CancelSimpleOrder(strategyId, status);
                await Task.Delay(500);

                await GetAllSimpleOrders();
                await Task.Delay(500);

                Console.WriteLine();
            }
            catch (Exception e)
            {
                Console.WriteLine("Error Running Example: " + e.Message);
            }
        }

        async Task<SimpleOrder> GetSimpleOrderById(string strategyId)
        {
            Console.WriteLine("*** GETTING SIMPLE ORDER BY ID ***");
            var response = await _httpClient.GetAsync($"simple-order/{strategyId}");
            response.EnsureSuccessStatusCode();

            var listContent = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
            var deserializedResponse = JsonSerializer.Deserialize<SimpleOrder>(listContent);
            Console.WriteLine("SIMPLE ORDER: " + JsonSerializer.Serialize(deserializedResponse));
            Console.WriteLine();

            return deserializedResponse;
        }

        async Task GetAllSimpleOrders(int page = 0)
        {
            Console.WriteLine("*** GETTING ALL SIMPLE ORDERS ***");
            var response = await _httpClient.GetAsync($"simple-order?page={page}");

            await ValidateResponses.ValidateGetResponse<SimpleOrder>(response);
        }

        async Task<string> NewSimpleOrder()
        {
            Console.WriteLine("*** CREATING NEW SIMPLE ORDER ***");
            var response = await _httpClient.PostAsJsonAsync("simple-order",
                new NewSimpleOrderRequest
                {
                    Broker = _config.Broker,
                    Account = _config.Account,
                    OrderType = SimpleOrderType.LIMIT,
                    Symbol = "SULA11",
                    Side = Side.BUY,
                    Quantity = 300,
                    Price = 15.45,
                    TimeInForce = TimeInForce.DAY
                });

            return await ValidateResponses.ValidateNewResponse(response);
        }

        async Task UpdateSimpleOrder(string strategyId, StrategyStatus status)
        {
            Console.WriteLine("*** UPDATING SIMPLE ORDER ***");
            if (SimpleOrderCanBeUpdated(status))
            {
                var response = await _httpClient.PutAsJsonAsync($"simple-order/{strategyId}",
                new UpdateSimpleOrderRequest { Quantity = 50000 });

                await ValidateResponses.ValidateUpdateCancelResponse(response);
            }
            else
            {
                Console.WriteLine("Simple order update cannot be done, since the current status is: " + status);
                Console.WriteLine();
            }
        }

        async Task CancelSimpleOrder(string strategyId, StrategyStatus status)
        {
            Console.WriteLine("*** CANCELLING SIMPLE ORDER ***");
            if (SimpleOrderCanBeUpdated(status))
            {
                var response = await _httpClient.DeleteAsync($"simple-order/{strategyId}");

                await ValidateResponses.ValidateUpdateCancelResponse(response);
            }
            else
            {
                Console.WriteLine("Simple order cancel cannot be done, since the current status is: " + status);
                Console.WriteLine();
            }
        }

        bool SimpleOrderCanBeUpdated(StrategyStatus status)
        {
            if (status == StrategyStatus.CANCELLED ||
                status == StrategyStatus.FINISHED ||
                status == StrategyStatus.TOTALLY_EXECUTED)
                return false;

            return true;
        }
    }
}

File: python/simple_order_example.py
from base_client import *

BROKER = '<BROKER>'   # Ex: '321'
ACCOUNT = '<ACCOUNT>' # Ex: '123'

new_request = {
    'Broker': BROKER,
    'Account': ACCOUNT,
    'OrderType': 'LIMIT',
    'Symbol': 'BBSE3',
    'Side': 'BUY',
    'Quantity': 1000,
    'TimeInForce': 'DAY',
    'Price': 20.04
}

update_request = {
    'Quantity': 20000
}


def main():

    access_token = get_token()
    headers = {
        'Content-Type': 'application/json',
        'Authorization': 'Bearer ' + access_token
    }

    client = BaseClient(headers, 'simple-order')

    print('*** CREATING SIMPLE ORDER ***')
    response = client.new(new_request)
    print(response.text)
    print()

    strategy_id = response.json()['StrategyId']

    print('*** GETTING ALL SIMPLE ORDERS ***')
    response = client.get(0)
    print(response.text)
    print()

    print('*** GETTING SIMPLE ORDER BY ID ***')
    response = client.get_by_id(strategy_id)
    status = response.json()['Status']
    print(response.text)
    print()

    if (client.is_order_updatable(status)):
        print('*** UPDATING SIMPLE ORDER ***')
        response = client.update(update_request, strategy_id)
        print(response.text)
        print()

        print('*** CANCELLING SIMPLE ORDER ***')
        response = client.cancel(strategy_id)
        print(response.text)
        print()

if __name__ == "__main__":
    main()
